File: Data Structures & Algorithms/combination-target-sum/submission-2.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        def backtrack(nums, slate, result, total, target, index):
            if total == target:
                result.append(slate.copy())
                return 
            
            for i in range(index, len(nums)):
                if total + nums[i] > target:
                    return
                slate.append(nums[i])
                backtrack(nums, slate, result, total + nums[i],target,  i)
                slate.pop()
        
        slate, result = [], []
        nums.sort()
        backtrack(nums, slate, result, 0, target, 0)
        return result

File: Data Structures & Algorithms/two-integer-sum/submission-2.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        var set = new Dictionary<int,int>();

        for(int i = 0; i< nums.Length; i++)
        {
            if(set.Keys.Contains(nums[i]))
            {
                return new int[] {set[nums[i]], i};
            }
            else
            {
                set.Add(target - nums[i], i);
            }
        }

        return null;
    }
}
